dali/test/python/test_fw_iterators.py

```diff
@@ -1051,6 +1052,33 @@ def get_data():
             counter += 1
     assert counter == iter_limit * runs
 
+def check_external_source_variable_size(Iterator, *args, iter_size=-1, **kwargs):
+    batch_size = 4
+    iter_limit = 4
+    runs = 3
+    test_data_shape = [2, 3, 4]
+    i = 0
+    def get_data():
+        nonlocal i
+        if i == iter_limit:
+            i = 0
+            raise StopIteration
+        out = [[np.random.randint(0, 255, size = test_data_shape, dtype = np.uint8) for _ in range(random.randint(1, batch_size))]]
+        i += 1
+        return out
```

Comment:
```suggestion
      dataset = [[[np.random.randint(0, 255, size = test_data_shape, dtype = np.uint8) for _ in range(random.randint(1, batch_size))]] for _ in range(iter_limit)]
```
This will allow you to actually compare outputs against this pre-generated data. Aslo, passing `dataset` as the `source` will raise StopIteration automatically.

Reply:
Not possible with the current ES behaviour.